DataFormats/CSCDigi/interface/CSCConstants.h

```diff
@@ -0,0 +1,133 @@
+#ifndef DataFormats_CSCDigi_CSCConstants_h
+#define DataFormats_CSCDigi_CSCConstants_h
+
+/**
+ * \class CSCConstants
+ *
+ * Static interface to basic chamber constants.
+ */
+
+class CSCConstants {
+public:
+  enum DDU_Info { NUM_DDUS = 5 };
+
+  enum CFEB_Info {
+    // Run-1: Maximum number of cathode front-end boards
+    MAX_CFEBS = 5,
+    // ME1/1 cases
+    NUM_CFEBS_ME1A_GANGED = 1,
+    NUM_CFEBS_ME1A_UNGANGED = 3,
+    NUM_CFEBS_ME1B = 4,
+    NUM_CFEBS_ME11_GANGED = NUM_CFEBS_ME1A_GANGED + NUM_CFEBS_ME1B,      // 5
+    NUM_CFEBS_ME11_UNGANGED = NUM_CFEBS_ME1A_UNGANGED + NUM_CFEBS_ME1B,  // 7
+    // Run-2: Maximum number of cathode front-end boards
+    MAX_CFEBS_RUN2 = NUM_CFEBS_ME11_UNGANGED,  // 7
+  };
+
+  enum FPGA_Latency { CLCT_EMUL_TIME_OFFSET = 3, ALCT_EMUL_TIME_OFFSET = 6 };
+
+  // Numbers obtained from https://twiki.cern.ch/twiki/pub/CMS/CSCDPGGeometry/table_of_csc_properties_150730.pdf
+  enum WG_Info {
+    NUM_WIREGROUPS_ME11 = 48,
+    NUM_WIREGROUPS_ME12 = 64,
+    NUM_WIREGROUPS_ME13 = 32,
+    NUM_WIREGROUPS_ME21 = 112,
+    NUM_WIREGROUPS_ME22 = 64,
+    NUM_WIREGROUPS_ME31 = 96,
+    NUM_WIREGROUPS_ME32 = 64,
+    NUM_WIREGROUPS_ME41 = 96,
+    NUM_WIREGROUPS_ME42 = 64,
+    // this number should really be 112, but has always been 119 since the
+    // CSC trigger was developed in 2006. Probably it would not hurt to change it to 112
+    MAX_NUM_WIREGROUPS = 119
+  };
+
+  // distrips, strips, half-strips
+  enum Strip_Info {
+    // Each CFEB reads out 8 distrips...
+    NUM_DISTRIPS_PER_CFEB = 8,
+    //...16 strips...
+    NUM_STRIPS_PER_CFEB = 2 * NUM_DISTRIPS_PER_CFEB,
+    //...32 half-strips.
+    NUM_HALF_STRIPS_PER_CFEB = 2 * NUM_STRIPS_PER_CFEB,
+    // There are exactly 80 or 112 strips...
+    NUM_STRIPS_5CFEBS = 5 * NUM_STRIPS_PER_CFEB,  // 80
+    NUM_STRIPS_7CFEBS = 7 * NUM_STRIPS_PER_CFEB,  // 112
+    //...and 160 or 224 half-strips for 5 or 7 CFEBs...
+    NUM_HALF_STRIPS_5CFEBS = 5 * NUM_HALF_STRIPS_PER_CFEB,  // 160
```

Comment:
```suggestion
    NUM_HALF_STRIPS_5CFEBS = MAX_CFEBS * NUM_HALF_STRIPS_PER_CFEB,  // 80
    NUM_HALF_STRIPS_7CFEBS = MAX_CFEBS_RUN2 * NUM_HALF_STRIPS_PER_CFEB,  // 112
```
And then, probably, to maintain the same nomenclature everywhere, these could be `NUM_HALF_STRIPS` and `NUM_HALF_STRIPS_RUN2`